x-pack/plugins/security_solution/public/sourcerer/experimental/components/dataview_picker/index.test.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { useKibana } from '../../../../common/lib/kibana/kibana_react';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectDataView } from '../../redux/actions';
+import { DataViewPicker } from '.';
+
+// Mock the required hooks and dependencies
+jest.mock('../../../../common/lib/kibana/kibana_react', () => ({
+  useKibana: jest.fn(),
+}));
+
+jest.mock('react-redux', () => ({
+  useDispatch: jest.fn(),
```

Comment:
One question @lgestc , Why do we need to mock Redux actions here. I think unit test of this components should tests all the custom actions for example, choose a different dataview or creating an adhoc dataview.

IMO, this test should only mock network requests and nothing else. Let me know what you think.

Reply:
I don't have a hard opinion about it. That kind of tests makes sure that your UI is changing correctly based on your Redux state.

Reply:
what do you mean? I am testing if actions are dispatched here.

Reply:
I meant that since this is the parent component, it should test the component e2e without mocking anything, not just the action dispatch.

Reply:
also by mocking the 3rd party component, we are making the tests faster to execute - since they dont pull in the entire dep tree just to see if the thing renders

Reply:
I agree 100% that they are faster and most of the test are like this only which test a small unit.

Even though we are using a third party component, we are passing custom behaviours to it so I am in the favour of having one e2e suite. But I will leave this decision on you for this particular component you know how must custom logic this is going to contain as you develop this.